libraries/lib-uuid/Uuid.cpp

```diff
@@ -0,0 +1,211 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file Uuid.cpp
+ @brief Define a class to generate and parse UUIDs
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+/*!********************************************************************
+
+ @class Uuid
+ @brief Platform independent class for generating and parsing UUIDs.
+
+ **********************************************************************/
+
+#include "Uuid.h"
+
+#include <algorithm>
+#include <cstring>
+
+#ifdef USE_UUID_CREATE
+#   include <rpc.h>
+#endif
+
+#ifdef USE_CFUUID
+#   include <CoreFoundation/CFUUID.h>
+#endif
+
+#ifdef USE_LIBUUID
+#   include <uuid/uuid.h>
+#endif
+
+#include "lib-string-utils/HexHelpers.h"
+
+namespace audacity
+{
+
+namespace
+{
+bool readByte (
+    Uuid::Bytes::iterator& outputIt, 
+    std::string::const_iterator& inputIt, 
+    const std::string::const_iterator& inputEnd)
+{
+    if (inputIt == inputEnd)
+        return false;
+
+    const char c1 = *inputIt++;
+
+    if (!IsHexChar (c1))
+        return false;
+
+    if (inputIt == inputEnd)
+        return false;
+
+    const char c2 = *inputIt++;
+
+    if (!IsHexChar (c2))
+        return false;
+
+    *outputIt = uint8_t (
+        (HexCharToNum (c1) << 4) | 
+        HexCharToNum (c2)
+    );
+
+    ++outputIt;
+
+    return true;
+}
+}
+
+Uuid::Uuid ()
+    : Uuid (Bytes {})
+{
+}
+
+Uuid::Uuid (const Bytes& data) noexcept
+    : mData (data)
+{
+}
+
+Uuid Uuid::Generate ()
```

Comment:
Are there any safeguards that `USE_UUID_CREATE` and `USE_LIBUUID` are not defined at the same time? Both of them seems to work on Windows. Does this function silently fail if none of these three libs are present?